refactor(core): Remove `afterRender`

JeanMeche opened the pull request:
During the `developerPreview` phase for the `afterRender` API we collected feedback and observed the usages that were made of it. 

We noticed that this API was often used in cases where better alternatives existed: 

* Native Observers (`MutationObserver`, `ResizeObserver`, `IntersectionObserver`)
* `afterRenderEffect` with behaves almost similarly but only if one of its signal depencies has changes
* `afterNextRender` when you actually need to schedule it only once.

Without strong usecases for a hook the always run after rendering (ie, directly tied to Angular's rendering cycle), we decided to remove this API. Since it was in developer preview, the [breaking change policy](https://angular.dev/reference/releases#breaking-change-policy-and-update-paths) does not apply.

wartab commented:
Guess that also fixes this issue https://github.com/angular/angular/issues/54825

JeanMeche commented:
If we indeed go in that direction, yes ! We're finishing collecting feedback on that topic.

JeanMeche commented:
Waiting for g3 to be cleaned up

josephperrott reviewed (approved):
LGTM

JeanMeche committed:
refactor(core): remove `afterRender` hook from the publicAPI

Feedback collected during developer preview did not surface a strong enough case to keep this API.

pkozlowski-opensource commented:
Going with #60951 instead.
